app/controllers/access_tokens_controller.rb

```diff
@@ -23,6 +23,15 @@ def create
     end
   end
 
+  def destroy
+    access_token = AccessToken.find_by token: params[:token]
+
+    render problem: { error: 'invalid_request' }, status: :bad_request and return unless access_token
```

Comment:
見つからなかった場合ならnot foundの方が適切かと思ったんですが、リクエストが変なのは確かなので、間違ったことを言っているわけではないんですよね　うーん

Reply:
この仕様は、アクセストークン無効化に関するRFCに従ったものです。

https://datatracker.ietf.org/doc/html/rfc7009#section-2.2.1

実際には、アクセストークン発行時のエラーレスポンスにプラスアルファしたものです。

> The error presentation conforms to the definition in Section 5.2 of
   [RFC6749].  The following additional error code is defined for the
   token revocation endpoint:

そして、アクセストークン発行時のエラーレスポンスでは、基本的に400を返すとあります。今回もこれに該当すると考えます。

> The authorization server responds with an HTTP 400 (Bad Request)
   status code (unless specified otherwise) and includes the following
   parameters with the response:

https://datatracker.ietf.org/doc/html/rfc6749#section-5.2

Reply:
というか、アクセストークンが発見できなかった場合でも、200を返すことになっているのを忘れていました。

> Note: invalid tokens do not cause an error response since the client
   cannot handle such an error in a reasonable way.  Moreover, the
   purpose of the revocation request, invalidating the particular token,
   is already achieved.

https://datatracker.ietf.org/doc/html/rfc7009#section-2.2

なので、今回の場合であれば、エラーレスポンスを返すのは、アクセストークンがパラメータに含まれていない場合のみになります。

Reply:
いや、これは不正なアクセストークンではなく、すでに無効化されているトークンという意味かな。

Reply:
400を返すのが正式な仕様だったんですね　なるほど